src/gui/qgsnewgeopackagelayerdialog.cpp

```diff
@@ -129,6 +130,17 @@ QgsNewGeoPackageLayerDialog::QgsNewGeoPackageLayerDialog( QWidget *parent, Qt::W
     }
     checkOk();
   } );
+
+  settings.beginGroup( "ogr/GPKG/connections", QgsSettings::Providers );
```

Comment:
Can you use the connections API instead of raw setting traversal?

Reply:
do you have a pointer to the API? which class is it? cheers

Reply:
https://qgis.org/api/classQgsProviderMetadata.html#af560b01841d55eb53b430e92bb7aaac7

Reply:
Actually - you could probably just create the completer directly using QgsProviderConnectionModel.